modules/ximgproc/include/opencv2/ximgproc/fourier_descriptors.hpp

```diff
@@ -0,0 +1,133 @@
+#define __OPENCV_FOURIERDESCRIPTORS_HPP__
+
+#include <opencv2/core.hpp>
+
+namespace cv {
+    namespace ximgproc {
+
+        //! @addtogroup shape
+        //! @{
+
+        /** @brief Abstract base class for ContourFitting algorithms.
+        */
+        class CV_EXPORTS ContourFitting : public Algorithm
+        {
+            int ctrSize;
+            int fdSize;
+            std::vector<std::complex<double> > b;
+            std::vector<std::complex<double> > a;
+            std::vector<double> frequence;
+            std::vector<double> rho, psi;
+            void frequencyInit();
+            void fAlpha(double x, double &fn, double &df);
+            double distance(std::complex<double> r, double alpha);
+            double  newtonRaphson(double x1, double x2);
+        public:
+            /** @brief Fit two closed curves using fourier descriptors. More details in @cite PersoonFu1977 and @cite BergerRaghunathan1998
+
+            @param ctr number of Fourier descriptors equal to number of contour points after resampling.
+            @param fd Contour defining second shape (Target).
+            @sa ContourFitting
+            */
+            ContourFitting(int ctr=1024,int fd=16):ctrSize(ctr),fdSize(fd){};
+            /** @brief Fit two closed curves using fourier descriptors. More details in @cite PersoonFu1977 and @cite BergerRaghunathan1998
+
+            @param src Contour defining first shape.
+            @param dst Contour defining second shape (Target).
+            @param _alphaPhiST : alpha=_alphaPhiST(0,0), phi=_alphaPhiST(0,1), s=_alphaPhiST(0,2), Tx=_alphaPhiST(0,3), Ty=_alphaPhiST(0,4)
+            @param dist distance between src and dst after matching.
+            @param fdContour false then src and dst are contours and true src and dst are fourier descriptors.
+            @sa estimateTransformation
+            */
+            void estimateTransformation(InputArray src, InputArray dst, OutputArray _alphaPhiST, double *dist=0,bool fdContour=false);
+            /** @brief set number of Fourier descriptors used in estimateTransformation
+
+            @param n number of Fourier descriptors equal to number of contour points after resampling.
+            @sa setCtrSize
+            */
+            void setCtrSize(int n);
+            /** @brief set number of Fourier descriptors when estimateTransformation used vector<Point>
+
+            @param n number of fourier descriptors used for optimal curve matching.
+            @sa setFDSize
+            */
+            void setFDSize(int n);
+            /**
+            @returns number of fourier descriptors
+            @sa getCtrSize
+            */
+            int getCtrSize() { return ctrSize; };
+            /**
+            @returns number of fourier descriptors used for optimal curve matching
+            @sa getFDSize
+            */
+            int getFDSize() { return fdSize; };
+        };
+        //! @addtogroup ximgproc_filters
```

Comment:
Why do you need this group here? I think all the code should be inside of `ximgproc_shape`.